cli/src/infragpt/tools.py

```
@@ -0,0 +1,192 @@
+        # Build parameters from function signature
+        properties = {}
+        required = []
+        
+        for param_name, param in sig.parameters.items():
+            param_type = "string"  # Default type
+            param_description = f"Parameter {param_name}"
+            param_default = None if param.default == inspect.Parameter.empty else param.default
+            
+            # Try to infer type from annotation
+            if param.annotation != inspect.Parameter.empty:
+                if param.annotation == str:
```

Ruff (0.12.2)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)